test/terraform/gke_test.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2020 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test
+
+import (
+	"flag"
+	"testing"
+	"time"
+
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/stretchr/testify/assert"
+)
+
+var project string
+
+func TestTerraformGKEInstallConfig(t *testing.T) {
+	t.Logf("Using project %s\n", project)
+	terraformOptions := &terraform.Options{
+		TerraformDir: "../../build/terraform/gke/",
+		Vars: map[string]interface{}{
+			"project":     project,
+			"name":        "terratest-cluster",
+			"values_file": "",
+		},
+	}
+
+	defer destroy(t, terraformOptions)
+
+	terraform.InitAndApply(t, terraformOptions)
+
+	output := terraform.Output(t, terraformOptions, "host")
+	assert.Contains(t, output, "https://")
+}
+
+func destroy(t *testing.T, options *terraform.Options) {
+	options.Targets = []string{"module.helm_agones.helm_release.agones"}
+	terraform.Destroy(t, options)
+	time.Sleep(30 * time.Second)
```

Comment:
Curious - is there a way we could poll to see if this is complete, rather than waiting for a period of time? Waiting is probably pretty fragile.

Reply:
Gentle bump on this review.

Reply:
Returning back to this as well

Reply:
Added the wait of `kubectl services -n agones-system` count would be 0 using terratest `K8S` module. Some updates to development `.tf` file was needed - creating kubeconfig file.

Reply:
There is no other options to poll the result or at least nothing else come to my mind. But previous approach build a "bridge" for creating more complex tests.